deepchecks/nlp/context.py

```diff
@@ -0,0 +1,238 @@
+# ----------------------------------------------------------------------------
+# Copyright (C) 2021-2022 Deepchecks (https://www.deepchecks.com)
+#
+# This file is part of Deepchecks.
+# Deepchecks is distributed under the terms of the GNU Affero General
+# Public License (version 3 or later).
+# You should have received a copy of the GNU Affero General Public License
+# along with Deepchecks.  If not, see <http://www.gnu.org/licenses/>.
+# ----------------------------------------------------------------------------
+#
+"""Module for base nlp context."""
+import collections
+import typing as t
+
+import numpy as np
+
+from deepchecks.core.task_type import TaskType
+from deepchecks.nlp.dataset import NLPData
+
+from deepchecks.core import CheckFailure, CheckResult, DatasetKind
+from deepchecks.core.errors import (DatasetValidationError, DeepchecksNotSupportedError, DeepchecksValueError,
+                                    ModelValidationError, ValidationError)
+
+
+__all__ = [
+    'Context',
+    'TNLPPred'
+]
+
+
+TClassPred = np.array
```

Comment:
Why array and not sequance?

Reply:
Also allow multiclass

Reply:
Any classification predictions are a 2d array (n_samples, n_classes), binary, multiclass, multilabel. 
I can allow a sequence of sequences of floats and cast internally, just thought that as it's a model output it's reasonable it'll come in some torch/np.array vector.